hydra-cluster/test/Test/EndToEndSpec.hs

```diff
@@ -112,7 +114,35 @@ withClusterTempDir name =
   withTempDir ("hydra-cluster-e2e-" <> name)
 
 spec :: Spec
-spec = around (showLogsOnFailure "EndToEndSpec") $
+spec = around (showLogsOnFailure "EndToEndSpec") $ do
+  it "End-to-end offline mode" $ \tracer -> do
+    withTempDir ("offline-mode-e2e") $ \tmpDir -> do
+      let networkId = Testnet (NetworkMagic 42) -- from defaultChainConfig
+      let startingState =
+            [ (Alice, lovelaceToValue 100_000_000)
+            , (Bob, lovelaceToValue 100_000_000)
+            ]
+      (aliceCardanoVk, aliceCardanoSk) <- keysFor Alice
+      (bobCardanoVk, _) <- keysFor Bob
+      offlineConfig <- offlineConfigFor startingState tmpDir networkId
```

Comment:
Should any arbitrary `UTxO` work to be loaded by the `offline` mode?

I think we should encode that (and make it visible) in this test by doing something like this inline:

```hs
initialUTxO <- generate $ do
  a <- genUTxOFor aliceCardanoVk
  b <- genUTxOFor bobCardanoVk
  pure $ a <> b
Aeson.encodeFile (tmpDir </> "utxo.json") initialUTxO
let offlineConfig = OfflineConfig
      { initialUTxOFile = tmpDir </> "utxo.json"
      , ledgerGenesisFile = Nothing
      }
``` 

Of course this will test some random UTxO each time (it's not a property test), but should already provide a better coverage than only a single ADA value (e.g. other assets)

Also, this is more clear in that the offline mode loads this file (one can read the test to know and maintain this, instead of all functions behind `offlineConfigFor`).